src/main/scala/is/hail/stats/package.scala

```diff
@@ -281,6 +281,20 @@ package object stats {
   // Returns the x for which p = Prob(Z^2 > x) with Z^2 a chi-squared RV with df degrees of freedom
   def inverseChiSquaredTail(df: Double, p: Double): Double = ChiSquare.quantile(p, df, false, false)
 
+  def rpois(lambda: Double): Int = new PoissonDistribution(lambda).sample()
+
+  def dpois(k: Int, lambda: Double): Double = new PoissonDistribution(lambda).probability(k)
+
+  def ppois(k: Int, lambda: Double, lowerTail: Boolean = true): Double = {
+    val p = new PoissonDistribution(lambda).cumulativeProbability(k)
+    if (lowerTail) p else 1d - p
```

Comment:
For example, the lowerTail versus upperTail should be handled internally with jdistlib parameters, otherwise you'll get catastrophic loss of precision here with very small p.  Look at the pnorm example